app/session.go

```diff
@@ -71,7 +71,7 @@ func (a *App) GetSession(token string) (*model.Session, *model.AppError) {
 
 	if session.Id == "" {
 		var nErr error
-		if session, nErr = a.Srv().Store.Session().Get(sqlstore.WithMaster(context.Background()), token); nErr == nil {
+		if session, nErr = a.srv.userService.GetSessionContext(sqlstore.WithMaster(context.Background()), token); nErr == nil {
```

Comment:
Now that this whole logic is moving to a separate package, it would be a good time to combine this whole logic inside `GetSession`. 

The whole logic is just querying the cache, and then querying the DB if it doesn't exist in the cache. IMO, the entire thing (the `if session.Id == ""` part) can be encapsulated inside `GetSession`.

0/5 on whether you want to do it in this PR or later.